### PR TITLE
Feature search component

### PR DESCRIPTION
**Pull request by yodalee**
I finally got the solution to search function. The solution is adding a search category, and the search text input only show when user select this category.
When user drag component into schematic, qucs will search all the components to find which component is right. I have to say this solution is not very good, quite waste resource.

### CODE REVIEW

**Comment by yodalee** [comment]
I just rebase this to most current (though no conflict actually)
@guitorri can you help me review this code? I have tested it and no problem encounter.

**Comment by yodalee** [comment]
Rebase done, I modify the layout excep -20 spacing.
`addSpacing(-20);`
since it makes qucs looks like this:
desktop2

File: qucs/qucs/qucs.cpp [hunk 1]
```diff
@@ -317,14 +317,22 @@ void QucsApp::initView()
...
   QWidget *CompGroup  = new QWidget();
   QVBoxLayout *CompGroupLayout = new QVBoxLayout();
+  QHBoxLayout *CompSearchLayout = new QHBoxLayout();
 
   CompChoose = new QComboBox(this);
   CompComps = new QListWidget(this);
   CompComps->setViewMode(QListView::IconMode);
   CompComps->setGridSize(QSize(110,90));
+  CompSearch = new QLineEdit(this);
+  CompSearch->setPlaceholderText(tr("Search Components:"));
```
Comment (guitorri) [inline qucs/qucs/qucs.cpp]: No ':' needed

Comment (yodalee) [inline qucs/qucs/qucs.cpp]: fix

**Comment by guitorri** [comment]
Merged in 6688617. Thank you!

**Pull request closed without merging**